src/averageAges.js

```diff
@@ -55,7 +68,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filterKid = people
+    .filter(person => people.find(mother => person.mother === mother.name));
+
+  const sortKid = !onlyWithSon ? filterKid : filterKid
+    .filter(kid => kid.sex === 'm');
+
+  const difference = sortKid
+    .reduce((sum, kid) =>
+      sum + (kid.born - people.find(mother =>
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

Better create variable `motherBorn` for example, this way code would be easier to read.